fedimint-cli/src/lib.rs

```diff
@@ -602,19 +603,28 @@ impl FedimintCli {
                     &config.federation_id(),
                 ),
                 config.to_owned(),
-                invite_code,
             )
             .await
             .map_err_cli_general()
     }
 
     async fn handle_command(&mut self, cli: Opts) -> CliOutputResult {
         match cli.command.clone() {
-            Command::InviteCode => {
+            Command::InviteCode { peer } => {
                 let db = cli.load_rocks_db().await?;
-                let invite_code = get_invite_code_from_db(&db)
+                let client_config = Client::get_config_from_db(&db)
                     .await
-                    .ok_or_cli_msg(CliErrorKind::GeneralFailure, "invite code not found")?;
+                    .ok_or_cli_msg(CliErrorKind::GeneralFailure, "client config code not found")?;
+
+                let invite_code = client_config
+                    .global
+                    .api_endpoints
+                    .get(&peer)
+                    .map(|peer_url| {
+                        InviteCode::new(peer_url.url.clone(), peer, client_config.federation_id())
+                    })
+                    .ok_or_cli_msg(CliErrorKind::GeneralFailure, "peer not found")?;
```

Comment:
If we merged #4297 first we wouldn't have to manually construct the invite here but could rather use `ClientConfig::invite_code(&Self, PeerId)`.

Reply:
I always intended to rebase on https://github.com/fedimint/fedimint/pull/4297, that's why this is still a draft.